packages/node/src/lib/retries.ts

```diff
@@ -0,0 +1,86 @@
+import { AxiosError, AxiosInstance } from 'axios';
+import axiosRetry, { isNetworkError } from 'axios-retry';
+import { v4 as uuid } from 'uuid';
+import { INovuConfiguration } from './novu.interface';
+
+const NON_IDEMPOTENT_METHODS = ['post', 'patch'];
+
+export function makeRetriable(
+  axios: AxiosInstance,
+  config?: INovuConfiguration
+) {
+  axios.interceptors.request.use((axiosConfig) => {
+    // don't attach idempotency key for idempotent methods
+    if (
+      axiosConfig.method &&
+      !NON_IDEMPOTENT_METHODS.includes(axiosConfig.method)
+    ) {
+      return axiosConfig;
+    }
+
+    const idempotencyKey = axiosConfig.headers['Idempotency-Key'];
+    // that means intercepted request is retried, so don't generate new idempotency key
+    if (idempotencyKey) {
+      return axiosConfig;
+    }
+
+    axiosConfig.headers['Idempotency-Key'] = uuid();
+
+    return axiosConfig;
+  });
```

Comment:
I'm using axios interceptor to attach idempotency key header to each request.

Idempotency key is attached only to `POST` and `PATCH` methods because they are **not** idempotent by definition.